lib/SILOptimizer/Mandatory/DIMemoryUseCollector.cpp

```diff
@@ -478,6 +497,28 @@ bool DIMemoryObjectInfo::isElementLetProperty(unsigned Element) const {
   return false;
 }
 
+SingleValueInstruction *DIMemoryObjectInfo::findAnyInitSelfValue() const  {
+  // If the object is 'self', return its uninitialized value.
+  if (isAnyInitSelf())
+    return getUninitializedValue();
+
+  // Otherwise we need to scan entry block to find mark_uninitialized
+  // instruction that belongs to `self`.
+
+  auto *BB = getFunction().getEntryBlock();
+  if (!BB)
+    return nullptr;
+
+  for (auto &I : *BB) {
+    SILInstruction *Inst = &I;
+    if (auto *MUI = dyn_cast<MarkUninitializedInst>(Inst)) {
+      if (!MUI->isVar())
```

Comment:
Also if you're going to refactor anyway, might really help to put a comment explaining that this `isVar` check is how we determine if it's self, and maybe point people to `isAnyInitSelf` for details.

Reply:
Sure, will do!

Reply:
✅